### PR TITLE
[CIGolang]new pre step to take non latest tag for oc inject

### PR DESCRIPTION
**Pull request by kuiwang02**
/cc @Xia-Zhao-rh @jianlinliu @jiajliu @shellyyang1989 

per https://github.com/openshift/release/pull/57024#issuecomment-2370123260, make this PR.

### CODE REVIEW

File: ci-operator/step-registry/openshift-extended/upgrade/prenonNoc/openshift-extended-upgrade-prenonNoc-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+ref:
+  as: openshift-extended-upgrade-prenonNoc
```
Comment (jianlinliu) [inline ci-operator/step-registry/openshift-extended/upgrade/prenonNoc/openshift-extended-upgrade-prenonNoc-ref.yaml:2]: The content of this step is the same as `openshift-extended-upgrade-pre`, the only difference is `cli: oc4pre`, right?

BTW, I wonder what means `nonNoc` ?

Comment (kuiwang02) [reply ci-operator/step-registry/openshift-extended/upgrade/prenonNoc/openshift-extended-upgrade-prenonNoc-ref.yaml:2]: yes.
nonNoc means non N oc
I can not make good step name, could you please provide one? thanks

Comment (jianlinliu) [reply ci-operator/step-registry/openshift-extended/upgrade/prenonNoc/openshift-extended-upgrade-prenonNoc-ref.yaml:2]: How about naming it `openshift-extended-upgrade-pre-custom-cli`, once that, maybe we also need to change `cli: oc4pre` to `cli: custom`

Comment (kuiwang02) [reply ci-operator/step-registry/openshift-extended/upgrade/prenonNoc/openshift-extended-upgrade-prenonNoc-ref.yaml]: ok, done

**Pull request merged**